### DISCUSSION TITLE
📔【操作系统】用户态与内核态

### DISCUSSION THREAD

**Discussion by imageslr** [imageslr/blog#71] [Announcements]
https://imageslr.github.io/2020/07/07/user-mode-kernel-mode.html 

 为何要区分用户态和内核态 最简单的运行程序的方式是“直接执行”，即直接在 CPU 上执行任意程序。直接执行的问题是： 如何限制代码行为？比如禁止：设置特殊寄存器的值、访问存储器的任意位置、I/O 请求、申请更多系统资源等 在运行这个程序的时候，如何切换到另一个程序？进程调度应该是 OS 才有的权限

**Comment by 0x3f3f3f** [comment]
sigsegv 无效的内存引用
